source/material_model/grain_size.cc

```diff
@@ -211,6 +211,51 @@ namespace aspect
 
 
 
+    template <int dim>
+    double
+    GrainSize<dim>::
+    moment_of_grain_size_distribution (const int n) const
```

Comment:
@gassmoeller says to make this into an anonymous function by deleting it from the header file and to hide it by putting it into an anontmous name space. 

```suggestion
namespace
  {
    double
    moment_of_grain_size_distribution (const int n) const
    [...]
   }
```